src/ucs/memory/memtype_cache.c

```diff
@@ -132,9 +132,10 @@ UCS_PROFILE_FUNC_VOID(ucs_memtype_cache_update_internal,
                              &region_list);
     ucs_list_for_each_safe(region, tmp, &region_list, list) {
         if (action == UCS_MEMTYPE_CACHE_ACTION_SET_MEMTYPE) {
-            if (region->mem_type == mem_type) {
+            if ((region->mem_type == mem_type) ||
+                (region->mem_type == UCS_MEMORY_TYPE_LAST)) {
```

Comment:
not sure this is right, what if the "real" memory type of the adjacent UNKNOWN region is not really the same?

Reply:
let's do not add them as UNKNOWN
we discussed offline today that we can call `cudaPointerGetAttributes` when scanning memory mapping from `/proc/self/maps` and check whether this buffer is a CUDA buffer or not and them add it to MemType Cache with `UCS_MEMORY_TYPE_CUDA`, wdyt?

Reply:
also, we can add them as `UCS_MEMORY_TYPE_CUDA` or `UCS_MEMORY_TYPE_CUDA_MANAGED` checking `isManaged` flag from attributes

Reply:
> we discussed offline today that we can call `cudaPointerGetAttributes` when scanning memory mapping from `/proc/self/maps` and check whether this buffer is a CUDA buffer or not and them add it to MemType Cache with `UCS_MEMORY_TYPE_CUDA`, wdyt?

Let's try that